python/pyarrow/__init__.py

```diff
@@ -30,7 +30,9 @@
 """
 
 import gc as _gc
+import importlib
 import os as _os
+import platform
```

Comment:
Nit: since we're prefixing other stdlib imports with an underscore, can you keep this convention here too?